File: libraries/hds/cmp_struc.f
      subroutine cmp_struc(struct, comp, struc, status)
*+
*  Name:
*     CMP_STRUC

*  Purpose:
*     Enquire if component is structure.

*  Language:
*     VAX Fortran

*  Invocation:
*     CALL CMP_STRUC(LOC, NAME, STRUC, STATUS)

*  Description:
*     An enquiry is made as to whether or not the structure component
*     is a structure object.

*  Arguments:
*     LOC=CHARACTER*(DAT__SZLOC)
*        Variable containing a locator associated with a structured
*        data object.
*     NAME=CHARACTER*(*)
*        Expression specifying the component name of a primitive
*        object contained in the structure.
*     STRUC=LOGICAL
*        Variable to receive whether the object is a structure.
*     STATUS=INTEGER
*        Variable holding the status value. If this variable is not
*        SAI__OK on input, the routine will return without action.
*        If the routine fails to complete, this variable will be
*        set to an appropriate error number.

*  Algorithm:
*     Get object locator using DAT_FIND and
*     DAT_STRUC.

*  Authors:
*     Jack Giddings (UCL::JRG)
*     {enter_new_authors_here}

*  History:
*     3-JAN-1983:  Original.  (UCL::JRG)
*     15-APR-1987:  Improved prologue layout (RAL::AJC)
*     {enter_further_changes_here}

*  Bugs:
*     {note_any_bugs_here}

*-

*  Global Constants:
      INCLUDE 'SAE_PAR'
      INCLUDE 'DAT_PAR'

*  Arguments Given:
      character*(*) struct		! Structure Locator
      character*(*) comp		! Component Name

*  Arguments Returned:
      logical struc			! Whether component is structure
*    Status return :
      integer status			! Status Return

*  Local Variables:
      character*(DAT__SZLOC) loc	! Component locator

*.


      if (status .eq. SAI__OK) then
         call dat_find(struct, comp, loc, status)
         if (status .ne. SAI__OK) then
            call cmp_erdsn(struct, comp, status)
         else
            call dat_struc(loc, struc, status)
            if (status .ne. SAI__OK) then
               call cmp_erdsn(struct, comp, status)
            endif
            call dat_annul(loc, status)
         endif
      endif

      end

File: libraries/hds/cmpStruc.c
#include "sae_par.h"
#include "hds.h"
#include "dat_par.h"

/*
*+
*  Name:
*     cmpStruc

*  Purpose:
*     Enquire if component is a structure.
*
*  Synopsis:
*     void cmpStruc( HDSLoc *strucloc, const char *comp, hdsbool_t *struc, int *status )

*  Description:
*     A enquiry is made as to whether a structure component is a structure.

*  Parameters:
*     strucloc
*        Locator associated with a structured data object.
*     comp
*        The component name of a primitive object contained in the structure.
*     struc
*        Whether or not the component is a structure.
*     status
*        Global status value.

*  Authors:
*     GSB: Graham Bell (EAO)

*  History:
*     07-MAY-2019 (GSB):
*        Original version, based on equivalent Fortran routine.
*-
*/

void cmpStruc( HDSLoc *strucloc, const char *comp, hdsbool_t *struc, int *status ) {
   HDSLoc *loc = NULL;
   if (*status != SAI__OK) {
      return;
   }
   datFind(strucloc, comp, &loc, status);
   datStruc(loc, struc, status);
   datAnnul(&loc, status);
}
